nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -6117,27 +6123,53 @@ protected Collection<AbstractMetricsRegistry> populateFlowMetrics() {
 
         // Get Connection Status Analytics (predictions, e.g.)
         Set<Connection> connections = controllerFacade.getFlowManager().findAllConnections();
-        for (Connection c : connections) {
-            // If a ResourceNotFoundException is thrown, analytics hasn't been enabled
+        Collection<Map<String, Long>> predictions = Collections.synchronizedList(new ArrayList<>());
+
+        final boolean analyticsEnabled = Boolean.parseBoolean(properties.getProperty(NiFiProperties.ANALYTICS_PREDICTION_ENABLED, Boolean.FALSE.toString()));
+
+        if (analyticsEnabled) {
+            if (parallelProcessingThreadPool == null) {
+                parallelProcessingThreadPool = createParallelProcessingThreadPool();
```

Comment:
See same note on the pull request for the main branch. In this case as well, the thread pool should be created when the ServiceFacade class is initialized.